xwiki-platform-distribution/xwiki-platform-distribution-flavor/xwiki-platform-distribution-flavor-test/xwiki-platform-distribution-flavor-test-ui/src/test/it/org/xwiki/test/ui/CompareVersionsTest.java

```diff
@@ -276,6 +279,64 @@ public void testUnifiedDiffShowsInlineChanges()
+    /**
+     * Tests navigation between versions.
+     */
+    public void testVersionNavigation()
+    {
+        String queryString = "viewer=changes&rev1=1.2&rev2=6.4";
+        getUtil().gotoPage(getTestClassName(), testPage.getMetaDataValue("page"), "view", queryString);
+        ChangesPane changesPane = new ChangesPane();
+
+        assertTrue(changesPane.getFromVersionSummary().startsWith("From version < 1.2 >"));
+        assertTrue(changesPane.getToVersionSummary().startsWith("To version < 6.4 >"));
+
+        // Cannot click on the arrow as it has a width and height of 0 (even if it's visible)
+        queryString = changesPane.getArrowLeftShift().getAttribute("href");
+        getUtil().gotoPage(queryString);
+        changesPane = new ChangesPane();
+        assertTrue(changesPane.getFromVersionSummary().startsWith("From version 1.1 >"));
+        assertTrue(changesPane.getToVersionSummary().startsWith("To version < 6.3 >"));
+        assertTrue(!getUtil().getDriver().hasElement(By.id("arrow-left-shift")));
+        assertTrue(getUtil().getDriver().hasElement(By.id("arrow-right-shift")));
+
+        queryString = changesPane.getArrowRightShift().getAttribute("href");
+        getUtil().gotoPage(queryString);
+        changesPane = new ChangesPane();
+        assertTrue(changesPane.getFromVersionSummary().startsWith("From version < 1.2 >"));
+        assertTrue(changesPane.getToVersionSummary().startsWith("To version < 6.4 >"));
+        assertTrue(getUtil().getDriver().hasElement(By.id("arrow-left-shift")));
+        assertTrue(getUtil().getDriver().hasElement(By.id("arrow-right-shift")));
+
+        changesPane.getArrowLeftFrom().click();
+        changesPane = new ChangesPane();
+        assertTrue(changesPane.getFromVersionSummary().startsWith("From version 1.1 >"));
+        assertTrue(changesPane.getToVersionSummary().startsWith("To version < 6.4 >"));
+        assertTrue(!getUtil().getDriver().hasElement(By.id("arrow-left-shift")));
+        assertTrue(getUtil().getDriver().hasElement(By.id("arrow-right-shift")));
+
+        changesPane.getArrowRightFrom().click();
+        changesPane = new ChangesPane();
+        assertTrue(changesPane.getFromVersionSummary().startsWith("From version < 1.2 >"));
+        assertTrue(changesPane.getToVersionSummary().startsWith("To version < 6.4 >"));
+        assertTrue(getUtil().getDriver().hasElement(By.id("arrow-left-shift")));
+        assertTrue(getUtil().getDriver().hasElement(By.id("arrow-right-shift")));
+
+        changesPane.getArrowRightTo().click();
+        changesPane = new ChangesPane();
+        assertTrue(changesPane.getFromVersionSummary().startsWith("From version < 1.2 >"));
+        assertTrue(changesPane.getToVersionSummary().startsWith("To version < 6.5"));
+        assertTrue(getUtil().getDriver().hasElement(By.id("arrow-left-shift")));
+        assertTrue(!getUtil().getDriver().hasElement(By.id("arrow-right-shift")));
+
+        changesPane.getArrowLeftTo().click();
```

Comment:
This should go in the pageobject (PO) since it's the responsbility of the PO to perform the actions.

Reply:
So I should create a `click` method for each arrows instead of the `get`?

Reply:
The general idea is that PO's job to hide technical details and it should expose hight level API, it should not even talk about arrows (might be something else later). For example I would not "create a click method for each arrows" but instead have method corresponding to what we want to do, things like: switchToPreviousOriginalVersion(), switchToNextVersion(), etc.

Reply:
Note that for me it's acceptable to use `clickXXX()` because that's an action you can do on the page represented by the PO (we do that in lots of places). What Thomas mentioned is an even higher level of abstraction that would require less maintenance should we replace the click by something else but I wouldn't consider that to be mandatory. It's also simpler to understand that `clickXXX()` represents a click in the UI so you know exactly what the API you use will do in your test.